runtime/buffer.h

```diff
@@ -204,6 +204,11 @@ class raw_buffer {
   // Make a pointer to a buffer with an allocation for the buffer in the same allocation.
   static raw_buffer_ptr make_allocated(std::size_t elem_size, std::size_t rank, const class dim* dims);
 
+  // Like make_allocated(), but the base pointer (for the elements) is an arbitrary, unowned address
+  // that is expected to remain valid for the life of this buffer. (A typical use case for this might
+  // be if the buffer storage is in a memory-mapped file.)
+  static raw_buffer_ptr make_allocated_external(std::size_t elem_size, std::size_t rank, const class dim* dims, void* base);
```

Comment:
I think the helper we should have here is just `make`, and it only takes `elem_size, rank`. We had exactly this function before: https://github.com/dsharlet/slinky/blob/ced5ae6a0689bb016e646f9269c4c5a958a05a76/runtime/buffer.h#L210-L211

I think this is simpler, and we don't need `dims` because we don't need to figure out how big the memory for the buffer data needs to be.

Reply:
And then we just set `base` directly rather than passing it in, presumably.